packages/api/amplify_api_dart/lib/src/decorators/web_socket_auth_utils.dart

```diff
@@ -36,10 +45,18 @@ Future<Uri> generateConnectionUri(
   );
   final encodedAuthHeaders =
       base64.encode(json.encode(authorizationHeaders).codeUnits);
-  final endpointUri = Uri.parse(
+  var endpointUri = Uri.parse(
     config.endpoint.replaceFirst('appsync-api', 'appsync-realtime-api'),
   );
-  return Uri(scheme: 'wss', host: endpointUri.host, path: 'graphql').replace(
+  var path = 'graphql';
+  // Check for custom domain and format URL as documented on https://docs.aws.amazon.com/appsync/latest/devguide/custom-domain-name.html.
+  final appSyncDomainRegExp = RegExp(_appSyncEndpointPattern);
+  if (!appSyncDomainRegExp.hasMatch(config.endpoint)) {
+    endpointUri = Uri.parse(config.endpoint);
+    path = 'graphql/realtime';
+  }
+
+  return Uri(scheme: 'wss', host: endpointUri.host, path: path).replace(
```

Comment:
Don't think you need the regex, tbh. You could use `host.endsWith('.amazonaws.com')` or pattern matching:

```suggestion
  final endpointUri = Uri.parse(
    config.endpoint.replaceFirst('appsync-api', 'appsync-realtime-api'),
  );
  // Check for custom domain and format URL as documented on https://docs.aws.amazon.com/appsync/latest/devguide/custom-domain-name.html.
  final isCustomDomain = switch (endpointUri.host.split('.')) {
    [..._, 'amazonaws', 'com'] => false,
    _ => true,
  };
  return Uri(
    scheme: 'wss',
    host: endpointUri.host,
    path: isCustomDomain ? '/graphql/realtime' : '/graphql',
  ).replace(
```

Reply:
To be safe you could remove the `replaceAll` line as well for custom domains

Reply:
imo I think the regex is easier to read. Plus, since it's lifted from the android impl we know it covers existing customers.

Reply:
I could go either way here but decided to err on side of simplicity in next revision here, tried to document w comments a little more, etc...